firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutor.java

```diff
@@ -0,0 +1,40 @@
+package com.example.firstapp;
+import android.os.SystemClock;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import org.junit.Test;
+
+public class CUJExecutor {
+
+  
+
+    /**
+     * Executes a CUJ with preparatory actions pre and measured actions post. Caches information for later use.
+     */
+
+    public static void executeCUJ(String[] preCUJ, String[] postCUJ) throws Exception {
+        String[] CUJ = new String[preCUJ.length + postCUJ.length];
+        if (preCUJ != null) {
+            System.arraycopy(preCUJ, 0, CUJ, 0, preCUJ.length);
+        }
+        System.arraycopy(postCUJ, 0, CUJ, preCUJ.length, postCUJ.length);
+        String pkg = CUJ[0];
```

Comment:
Feels odd to be passing in the package as an action.  Instead, it should be its own separate argument.

Reply:
Idea behind this was that if the package is an action, user can choose whether to put it in the preparatory or measured actions list. Otherwise, I'd also need them to specify whether they want startup to be recorded/measured. Let me know what you think

Reply:
Right, but the arguments should not be simple Strings.  They should be objects and types representing what it is.